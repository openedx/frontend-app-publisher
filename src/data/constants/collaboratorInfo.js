export const CREATE_COLLABORATOR = 'CREATE_COLLABORATOR';
export const CREATE_COLLABORATOR_SUCCESS = 'CREATE_COLLABORATOR_SUCCESS';
export const CREATE_COLLABORATOR_FAIL = 'CREATE_COLLABORATOR_FAIL';
export const EDIT_COLLABORATOR_INFO = 'EDIT_COLLABORATOR_INFO';
export const EDIT_COLLABORATOR_INFO_SUCCESS = 'EDIT_COLLABORATOR_INFO_SUCCESS';
export const EDIT_COLLABORATOR_INFO_FAIL = 'EDIT_COLLABORATOR_INFO_FAIL';
export const CANCEL_COLLABORATOR_INFO = 'CANCEL_COLLABORATOR_INFO';
export const RESET_COLLABORATOR_INFO = 'RESET_COLLABORATOR_INFO';
