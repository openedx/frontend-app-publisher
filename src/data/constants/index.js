const VERIFIED_TRACK = Object.freeze({ name: 'Verified and Audit', key: 'verified' });
const AUDIT_TRACK = Object.freeze({ name: 'Audit Only', key: 'audit' });
const PROFESSIONAL_TRACK = Object.freeze({ name: 'Professional Only', key: 'professional' });
const MASTERS_TRACK = Object.freeze({ key: 'masters' });
const ENTITLEMENT_TRACKS = Object.freeze(['verified', 'professional']);

const REVIEW_BY_LEGAL = 'review_by_legal';
const REVIEW_BY_INTERNAL = 'review_by_internal';
const PUBLISHED = 'published';
const REVIEWED = 'reviewed';
const UNPUBLISHED = 'unpublished';
const ARCHIVED = 'archived';
const IN_REVIEW_STATUS = Object.freeze([REVIEW_BY_LEGAL, REVIEW_BY_INTERNAL]);
const POST_REVIEW_STATUSES = Object.freeze([PUBLISHED, REVIEWED]);
const COURSE_EXEMPT_FIELDS = Object.freeze(['imgSrc', 'videoSrc']);
const COURSE_RUN_NON_EXEMPT_FIELDS = Object.freeze(['expected_program_name', 'expected_program_type']);

const DATE_FORMAT = 'YYYY-MM-DD[T]HH:mm:ss[Z]';
const DATE_INPUT_PATTERN = '20[1-9][0-9]/(0[1-9]|1[012])/(0[1-9]|[12][0-9]|3[01])';
const FORMAT_DATE_MATCHER = /20\d{2}-(0\d{1}|1[0-2])-([0-2]\d{1}|3[0-1])/;
const NORMALIZE_DATE_MATCHER = /20\d{2}\/(0\d{1}|1[0-2])\/([0-2]\d{1}|3[0-1])/;

const EXECUTIVE_EDUCATION_SLUG = 'executive-education-2u';
const BOOTCAMP_SLUG = 'bootcamp-2u';

const COURSE_URL_SLUG_PATTERN_OLD = '^[a-z0-9_]+(?:-[a-z0-9_]+)*$';
const COURSE_URL_SLUG_PATTERN_NEW = '^learn/[a-z0-9_]+(?:-?[a-z0-9_]+)*/[a-z0-9_]+(?:-?[a-z0-9_]+)*$';
const COURSE_URL_SLUG_PATTERN = `${COURSE_URL_SLUG_PATTERN_NEW}|${COURSE_URL_SLUG_PATTERN_OLD}`;

const COURSE_URL_SLUG_VALIDATION_MESSAGE = {
  [COURSE_URL_SLUG_PATTERN_OLD]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only.',
  [COURSE_URL_SLUG_PATTERN_NEW]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only and must be in the format learn/<primary_subject>/<org-slug>-<course_slug>',
  [COURSE_URL_SLUG_PATTERN]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only and must be in the format <custom-url-slug> or learn/<primary_subject>/<org-slug>-<course_slug>.',
};

export {
  VERIFIED_TRACK,
  AUDIT_TRACK,
  PROFESSIONAL_TRACK,
  MASTERS_TRACK,
  ENTITLEMENT_TRACKS,
  REVIEW_BY_LEGAL,
  REVIEW_BY_INTERNAL,
  PUBLISHED,
  REVIEWED,
  UNPUBLISHED,
  ARCHIVED,
  IN_REVIEW_STATUS,
  POST_REVIEW_STATUSES,
  DATE_FORMAT,
  DATE_INPUT_PATTERN,
  FORMAT_DATE_MATCHER,
  NORMALIZE_DATE_MATCHER,
  COURSE_EXEMPT_FIELDS,
  COURSE_RUN_NON_EXEMPT_FIELDS,
  EXECUTIVE_EDUCATION_SLUG,
  BOOTCAMP_SLUG,
  COURSE_URL_SLUG_PATTERN,
  COURSE_URL_SLUG_PATTERN_NEW,
  COURSE_URL_SLUG_PATTERN_OLD,
  COURSE_URL_SLUG_VALIDATION_MESSAGE,
};
