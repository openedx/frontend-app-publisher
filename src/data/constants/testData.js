const courseOptions = {
  data: {
    actions: {
      POST: {
        type: {
          type_options: [
            {
              uuid: '9521aa7d-801b-4a67-92c3-716ea30f5086',
              name: 'Credit',
              course_run_types: [
                {
                  uuid: 'f17e29d6-4648-4bb5-a199-97dc40f904aa',
                  name: 'Credit',
                  modes: ['credit', 'verified', 'audit'],
                  is_marketable: true,
                },
                {
                  uuid: '4e260c57-24ef-46c1-9a0d-5ec3a30f6b0c',
                  name: 'Verified and Audit',
                  modes: ['verified', 'audit'],
                  is_marketable: true,
                },
                {
                  uuid: 'cfacfc62-54bd-4e1b-939a-5a94f12fbd8d',
                  name: 'Audit Only',
                  modes: ['audit'],
                  is_marketable: true,
                },
              ],
              entitlement_types: ['verified'],
              tracks: [
                {
                  seat_type: {
                    name: 'Credit',
                    slug: 'credit',
                  },
                  mode: {
                    name: 'Credit',
                    slug: 'credit',
                    is_id_verified: true,
                    is_credit_eligible: true,
                    certificate_type: 'credit',
                    payee: 'platform',
                  },
                },
                {
                  seat_type: {
                    name: 'Verified',
                    slug: 'verified',
                  },
                  mode: {
                    name: 'Verified',
                    slug: 'verified',
                    is_id_verified: true,
                    is_credit_eligible: false,
                    certificate_type: 'verified',
                    payee: 'platform',
                  },
                },
                {
                  seat_type: {
                    name: 'Audit',
                    slug: 'audit',
                  },
                  mode: {
                    name: 'Audit',
                    slug: 'audit',
                    is_id_verified: false,
                    is_credit_eligible: false,
                    certificate_type: '',
                    payee: '',
                  },
                },
              ],
            },
            {
              uuid: '7b41992e-f268-4331-8ba9-72acb0880454',
              name: 'Masters Only',
              course_run_types: [
                {
                  uuid: 'f394732f-ba43-4260-8ada-06a9f18e7160',
                  name: 'Masters Only',
                  modes: ['masters'],
                  is_marketable: false,
                },
              ],
              entitlement_types: [],
              tracks: [
                {
                  seat_type: null,
                  mode: {
                    name: 'Masters',
                    slug: 'masters',
                    is_id_verified: false,
                    is_credit_eligible: false,
                    certificate_type: '',
                    payee: 'organization',
                  },
                },
              ],
            },
            {
              uuid: '8a8f30e1-23ce-4ed3-a361-1325c656b67b',
              name: 'Verified and Audit',
              course_run_types: [
                {
                  uuid: '4e260c57-24ef-46c1-9a0d-5ec3a30f6b0c',
                  name: 'Verified and Audit',
                  modes: ['verified', 'audit'],
                  is_marketable: true,
                },
                {
                  uuid: 'cfacfc62-54bd-4e1b-939a-5a94f12fbd8d',
                  name: 'Audit Only',
                  modes: ['audit'],
                  is_marketable: true,
                },
              ],
              entitlement_types: ['verified'],
              tracks: [
                {
                  seat_type: {
                    name: 'Verified',
                    slug: 'verified',
                  },
                  mode: {
                    name: 'Verified',
                    slug: 'verified',
                    is_id_verified: true,
                    is_credit_eligible: false,
                    certificate_type: 'verified',
                    payee: 'platform',
                  },
                },
                {
                  seat_type: {
                    name: 'Audit',
                    slug: 'audit',
                  },
                  mode: {
                    name: 'Audit',
                    slug: 'audit',
                    is_id_verified: false,
                    is_credit_eligible: false,
                    certificate_type: '',
                    payee: '',
                  },
                },
              ],
            },
            {
              uuid: 'fdde7d04-7ce0-4ff7-ac90-7f518e90ac1e',
              name: 'Professional Only',
              course_run_types: [
                {
                  uuid: '6cebfc50-abca-4674-b4c7-249b31af6ddb',
                  name: 'Professional Only',
                  modes: ['professional'],
                  is_marketable: true,
                },
              ],
              entitlement_types: ['professional'],
              tracks: [
                {
                  seat_type: {
                    name: 'Professional',
                    slug: 'professional',
                  },
                  mode: {
                    name: 'Professional',
                    slug: 'professional',
                    is_id_verified: true,
                    is_credit_eligible: false,
                    certificate_type: 'professional',
                    payee: 'platform',
                  },
                },
              ],
            },
            {
              uuid: '03e09c15-4127-4031-bc02-e20fcbdf09f0',
              name: 'Audit Only',
              course_run_types: [
                {
                  uuid: 'cfacfc62-54bd-4e1b-939a-5a94f12fbd8d',
                  name: 'Audit Only',
                  modes: ['audit'],
                  is_marketable: true,
                },
              ],
              entitlement_types: ['audit'],
              tracks: [
                {
                  seat_type: {
                    name: 'Audit',
                    slug: 'audit',
                  },
                  mode: {
                    name: 'Audit',
                    slug: 'audit',
                    is_id_verified: false,
                    is_credit_eligible: false,
                    certificate_type: '',
                    payee: '',
                  },
                },
              ],
            },
          ],
        },
        level_type: {
          choices: [
            { display_name: 'Beginner', value: 'beginner' },
            { display_name: 'Intermediate', value: 'intermediate' },
            { display_name: 'Advanced', value: 'advanced' },
          ],
        },
        subjects: {
          child: {
            choices: [
              { display_name: 'Business', value: 'business' },
              { display_name: 'Chemistry', value: 'chemistry' },
              { display_name: 'English', value: 'english' },
              { display_name: 'Security', value: 'security' },
            ],
          },
        },
        additional_metadata: {
          children: {
            product_status: {
              choices: [
                { display_name: 'Published', value: 'published' },
                { display_name: 'Archived', value: 'archived' },
              ],
            },
            external_course_marketing_type: {
              choices: [
                { display_name: 'Short Course', value: 'short_course' },
                { display_name: 'Sprint', value: 'sprint' },
              ],
            },
          },
        },
        location_restriction: {
          children: {
            restriction_type: {
              choices: [
                'blocklist',
                'allowlist',
              ],
            },
            countries: {
              child: {
                choices: [
                  { label: 'Afghanistan', value: 'AF' },
                  { label: 'Ã…land Islands', value: 'AX' },
                  { label: 'Algeria', value: 'DZ' },
                  { label: 'American Samoa', value: 'AS' },
                  { label: 'Andorra', value: 'AD' },
                ],
              },
            },
            states: {
              child: {
                choices: [
                  { label: 'Alabama', value: 'AL' },
                  { label: 'Arizona', value: 'AZ' },
                  { label: 'Arkansas', value: 'AR' },
                  { label: 'California', value: 'CA' },
                  { label: 'Colorado', value: 'CO' },
                ],
              },
            },
          },
        },
      },
    },
  },
  isFetching: false,
  error: null,
};

const courseRunOptions = {
  data: {
    actions: {
      POST: {
        pacing_type: {
          type: 'choice',
          required: false,
          read_only: false,
          label: 'Pacing type',
          choices: [{
            display_name: 'Instructor-paced',
            value: 'instructor_paced',
          }, {
            display_name: 'Self-paced',
            value: 'self_paced',
          }],
        },
        content_language: {
          type: 'field',
          required: false,
          read_only: false,
          label: 'Content language',
          help_text: 'Language in which the course is administered',
          choices: [{
            display_name: 'Afrikaans',
            value: 'af',
          }, {
            display_name: 'Arabic - United Arab Emirates',
            value: 'ar-ae',
          }],
        },
        expected_program_type: {
          type: 'field',
          required: false,
          read_only: false,
          label: 'Expected Program Type',
          choices: [{
            display_name: 'Professional Certificate',
            value: 'professional-certificate',
          },
          {
            display_name: 'MicroMasters',
            value: 'micromasters',
          },
          {
            display_name: 'XSeries',
            value: 'xseries',
          },
          {
            display_name: 'Masters',
            value: 'masters',
          }],
        },
      },
    },
  },
  isFetching: false,
  error: null,
};

const collaboratorOptions = {
  data: {
    results: [],
  },
  error: null,
  isFetching: false,
};

export {
  courseOptions,
  courseRunOptions,
  collaboratorOptions,
};
