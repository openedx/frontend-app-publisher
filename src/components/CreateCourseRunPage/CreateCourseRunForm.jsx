import moment from 'moment';
import React from 'react';
import { Field, reduxForm } from 'redux-form';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import ButtonToolbar from '../ButtonToolbar';
import FieldLabel from '../FieldLabel';
import ActionButton from '../ActionButton';

import {
  endDateHelp, runTypeHelp, startDateHelp, pacingHelp, keyHelp,
} from '../../helpText';
import RenderInputTextField from '../RenderInputTextField';
import RenderSelectField from '../RenderSelectField';
import DateTimeField from '../DateTimeField';
import {
  getDateWithDashes, getOptionsData, isSafari, localTimeZone, parseOptions,
} from '../../utils';
import { DATE_INPUT_PATTERN } from '../../data/constants';

const BaseCreateCourseRunForm = ({
  handleSubmit,
  pristine,
  isCreating,
  title,
  uuid,
  courseRunOptions,
  courseRunTypeOptions,
  currentFormValues,
  courseRunLabels,
  canSetRunKey,
}) => {
  const courseRunOptionsData = getOptionsData(courseRunOptions);
  const { pacing_type: { choices } } = courseRunOptionsData;
  const pacingTypeOptions = courseRunOptionsData && parseOptions(choices);
  return (
    <div className="create-course-run-form">
      <h2>Create a new course run</h2>
      <hr />
      <FieldLabel text="Course" className="mb-2" />
      <div className="mb-3">
        <Link to={`/courses/${uuid}`}>
          {title}
        </Link>
      </div>
      <form onSubmit={handleSubmit}>
        <Field
          name="rerun"
          type="text"
          component={RenderSelectField}
          options={courseRunLabels}
          label={(
            <FieldLabel
              id="rerun.label"
              text="Select a run to copy"
              helpText={(
                <div>
                  <p>
                    Select a run to copy as the starting point for your new studio instance.
                    The newest run is listed first.
                  </p>
                </div>
              )}
            />
          )}
        />
        {canSetRunKey
        && (
        <Field
          name="courseRunKey"
          component={RenderInputTextField}
          type="text"
          pattern="[a-zA-Z0-9-]+"
          label={(
            <FieldLabel
              id="courseRunKey-label"
              text="Run Key"
              helpText={keyHelp}
              optional
            />
          )}
        />
        )}
        {/* TODO this should be refactored when paragon supports safari */}
        {/* text inputs for safari */}
        {isSafari
          ? (
            <div>
              <Field
                name="start"
                type="text"
                component={DateTimeField}
                dateLabel="Start date"
                timeLabel={`Start time (${localTimeZone})`}
                helpText={startDateHelp}
                required
                maxLength="10"
                pattern={DATE_INPUT_PATTERN}
                placeholder="yyyy/mm/dd"
              />
              <Field
                name="end"
                type="text"
                component={DateTimeField}
                dateLabel="End date"
                timeLabel={`End time (${localTimeZone})`}
                helpText={endDateHelp}
                required
                maxLength="10"
                pattern={DATE_INPUT_PATTERN}
                placeholder="yyyy/mm/dd"
              />
            </div>
          )
          // date inputs for all browsers besides safari
          : (
            <div>
              <Field
                name="start"
                type="date"
                component={DateTimeField}
                dateLabel="Start date"
                timeLabel={`Start time (${localTimeZone})`}
                helpText={startDateHelp}
                required
                minDate={getDateWithDashes(moment())}
              />
              <Field
                name="end"
                type="date"
                component={DateTimeField}
                dateLabel="End date"
                timeLabel={`End time (${localTimeZone})`}
                helpText={endDateHelp}
                required
                minDate={getDateWithDashes(moment(currentFormValues.start).add(1, 'd') || moment())}
              />
            </div>
          )}
        <Field
          name="run_type"
          component={RenderSelectField}
          options={courseRunTypeOptions}
          label={(
            <FieldLabel
              id="run_type.label"
              text="Course run enrollment track"
              required
              helpText={runTypeHelp}
            />
          )}
          required
        />
        <Field
          name="pacing_type"
          type="text"
          component={RenderSelectField}
          options={pacingTypeOptions}
          label={(
            <FieldLabel
              id="pacing_type.label"
              text="Course pacing"
              helpText={pacingHelp}
            />
          )}
        />
        <ButtonToolbar>
          <Link to={`/courses/${uuid}`}>
            <button
              type="button"
              className="btn btn-outline-primary"
              disabled={isCreating}
            >
              Cancel
            </button>
          </Link>
          <ActionButton
            disabled={pristine}
            labels={{
              default: 'Create',
              pending: 'Creating',
            }}
            state={isCreating ? 'pending' : 'default'}
          />
        </ButtonToolbar>
      </form>
    </div>
  );
};

BaseCreateCourseRunForm.defaultProps = {
  currentFormValues: {},
  courseRunLabels: [],
  courseRunOptions: {},
  courseRunTypeOptions: [],
};

BaseCreateCourseRunForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  uuid: PropTypes.string.isRequired,
  pristine: PropTypes.bool.isRequired,
  isCreating: PropTypes.bool.isRequired,
  currentFormValues: PropTypes.shape({
    start: PropTypes.string,
  }),
  courseRunLabels: PropTypes.arrayOf(PropTypes.shape({})),
  courseRunOptions: PropTypes.shape({
    data: PropTypes.shape(),
    error: PropTypes.arrayOf(PropTypes.string),
    isFetching: PropTypes.bool,
  }),
  courseRunTypeOptions: PropTypes.arrayOf(PropTypes.shape({})),
  canSetRunKey: PropTypes.bool.isRequired,
};

const CreateCourseRunForm = reduxForm({
  form: 'create-course-run-form',
})(BaseCreateCourseRunForm);
export { BaseCreateCourseRunForm, CreateCourseRunForm };
