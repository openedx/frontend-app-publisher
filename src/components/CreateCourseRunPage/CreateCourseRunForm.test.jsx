import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { BaseCreateCourseRunForm } from './CreateCourseRunForm';
import { courseOptions, courseRunOptions } from '../../data/constants/testData';
import { getOptionsData, parseCourseTypeOptions } from '../../utils';

Date.now = jest.fn(() => new Date(Date.UTC(2001, 0, 1)).valueOf());

describe('CreateCourseRunForm', () => {
  it('renders html correctly', () => {
    const component = shallow(<BaseCreateCourseRunForm
      handleSubmit={() => {}}
      initialValues={{
        course: 'edx+test101',
      }}
      title="Test Course"
      uuid="00000000-0000-0000-0000-000000000001"
      pristine
      isCreating={false}
      courseRunOptions={courseRunOptions}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders html correctly with Course Type', () => {
    const courseTypeUuid = '8a8f30e1-23ce-4ed3-a361-1325c656b67b';
    const courseOptionsData = getOptionsData(courseOptions);
    const parsedTypeOptions = parseCourseTypeOptions(courseOptionsData.type.type_options);
    const { courseRunTypeOptions } = parsedTypeOptions;
    const component = shallow(<BaseCreateCourseRunForm
      handleSubmit={() => {}}
      initialValues={{
        course: 'edx+test101',
      }}
      title="Test Course"
      uuid="00000000-0000-0000-0000-000000000001"
      pristine
      isCreating={false}
      courseRunTypeOptions={courseRunTypeOptions[courseTypeUuid]}
      courseRunOptions={courseRunOptions}
      courseTypeUuid={courseTypeUuid}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders html correctly when creating', () => {
    const component = shallow(<BaseCreateCourseRunForm
      handleSubmit={() => {}}
      initialValues={{
        course: 'edx+test101',
      }}
      title="Test Course"
      uuid="00000000-0000-0000-0000-000000000001"
      pristine={false}
      isCreating
      courseRunOptions={courseRunOptions}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
