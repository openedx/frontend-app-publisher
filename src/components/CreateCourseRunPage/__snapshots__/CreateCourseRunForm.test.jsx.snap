// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCourseRunForm renders html correctly 1`] = `
<div
  className="create-course-run-form"
>
  <h2>
    Create a new course run
  </h2>
  <hr />
  <FieldLabel
    className="mb-2"
    extraText=""
    helpText=""
    id={null}
    optional={false}
    text="Course"
  />
  <div
    className="mb-3"
  >
    <Link
      to="/courses/00000000-0000-0000-0000-000000000001"
    >
      Test Course
    </Link>
  </div>
  <form
    onSubmit={[Function]}
  >
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                Select a run to copy as the starting point for your new studio instance. The newest run is listed first.
              </p>
            </div>
          }
          id="rerun.label"
          optional={false}
          text="Select a run to copy"
        />
      }
      name="rerun"
      options={Array []}
      type="text"
    />
    <div>
      <Field
        component={[Function]}
        dateLabel="Start date"
        helpText={
          <div>
            <p>
              Required Format: yyyy/mm/dd
            </p>
            <p>
              Start on a Tuesday, Wednesday, or Thursday.
            </p>
            <p>
              Avoid major holidays.
            </p>
            <p>
              Dates are editable in Studio after the course is created, rounded to the nearest term. If your courses start on January - May 2020, the URL will end in 1T2020.
            </p>
            <p>
              If you are unsure of the exact date, specify a day that is close to the estimated start date. For example, if your course will start near the end of March, specify March 31.
            </p>
          </div>
        }
        maxLength="10"
        name="start"
        pattern="20[1-9][0-9]/(0[1-9]|1[012])/(0[1-9]|[12][0-9]|3[01])"
        placeholder="yyyy/mm/dd"
        required={true}
        timeLabel="Start time (GMT)"
        type="text"
      />
      <Field
        component={[Function]}
        dateLabel="End date"
        helpText={
          <div>
            <p>
              Required Format: yyyy/mm/dd
            </p>
            <p>
              If you are unsure of the exact date, specify a day that is close to the estimated end date. For example, if your course will end near the end of March, specify March 31.
            </p>
          </div>
        }
        maxLength="10"
        name="end"
        pattern="20[1-9][0-9]/(0[1-9]|1[012])/(0[1-9]|[12][0-9]|3[01])"
        placeholder="yyyy/mm/dd"
        required={true}
        timeLabel="End time (GMT)"
        type="text"
      />
    </div>
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                The enrollment track determines whether a course run offers a paid certificate and what sort of verification is required.
              </p>
              <p>
                <a
                  href="https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/glossary.html#enrollment-track-g"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Learn more.
                </a>
              </p>
            </div>
          }
          id="run_type.label"
          optional={false}
          required={true}
          text="Course run enrollment track"
        />
      }
      name="run_type"
      options={Array []}
      required={true}
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                Instructor-paced courses include individual assignments that have specific due dates before the course end date.
              </p>
              <p>
                Self-paced courses do not have individual assignments that have specific due dates before the course end date. All assignments are due on the course end date.
              </p>
            </div>
          }
          id="pacing_type.label"
          optional={false}
          text="Course pacing"
        />
      }
      name="pacing_type"
      options={
        Array [
          Object {
            "label": "Instructor-paced",
            "value": "instructor_paced",
          },
          Object {
            "label": "Self-paced",
            "value": "self_paced",
          },
        ]
      }
      type="text"
    />
    <ButtonToolbar
      className=""
      leftJustify={false}
    >
      <Link
        to="/courses/00000000-0000-0000-0000-000000000001"
      >
        <button
          className="btn btn-outline-primary"
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </Link>
      <ActionButton
        className=""
        disabled={true}
        labels={
          Object {
            "default": "Create",
            "pending": "Creating",
          }
        }
        primary={true}
        state="default"
      />
    </ButtonToolbar>
  </form>
</div>
`;

exports[`CreateCourseRunForm renders html correctly when creating 1`] = `
<div
  className="create-course-run-form"
>
  <h2>
    Create a new course run
  </h2>
  <hr />
  <FieldLabel
    className="mb-2"
    extraText=""
    helpText=""
    id={null}
    optional={false}
    text="Course"
  />
  <div
    className="mb-3"
  >
    <Link
      to="/courses/00000000-0000-0000-0000-000000000001"
    >
      Test Course
    </Link>
  </div>
  <form
    onSubmit={[Function]}
  >
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                Select a run to copy as the starting point for your new studio instance. The newest run is listed first.
              </p>
            </div>
          }
          id="rerun.label"
          optional={false}
          text="Select a run to copy"
        />
      }
      name="rerun"
      options={Array []}
      type="text"
    />
    <div>
      <Field
        component={[Function]}
        dateLabel="Start date"
        helpText={
          <div>
            <p>
              Required Format: yyyy/mm/dd
            </p>
            <p>
              Start on a Tuesday, Wednesday, or Thursday.
            </p>
            <p>
              Avoid major holidays.
            </p>
            <p>
              Dates are editable in Studio after the course is created, rounded to the nearest term. If your courses start on January - May 2020, the URL will end in 1T2020.
            </p>
            <p>
              If you are unsure of the exact date, specify a day that is close to the estimated start date. For example, if your course will start near the end of March, specify March 31.
            </p>
          </div>
        }
        maxLength="10"
        name="start"
        pattern="20[1-9][0-9]/(0[1-9]|1[012])/(0[1-9]|[12][0-9]|3[01])"
        placeholder="yyyy/mm/dd"
        required={true}
        timeLabel="Start time (GMT)"
        type="text"
      />
      <Field
        component={[Function]}
        dateLabel="End date"
        helpText={
          <div>
            <p>
              Required Format: yyyy/mm/dd
            </p>
            <p>
              If you are unsure of the exact date, specify a day that is close to the estimated end date. For example, if your course will end near the end of March, specify March 31.
            </p>
          </div>
        }
        maxLength="10"
        name="end"
        pattern="20[1-9][0-9]/(0[1-9]|1[012])/(0[1-9]|[12][0-9]|3[01])"
        placeholder="yyyy/mm/dd"
        required={true}
        timeLabel="End time (GMT)"
        type="text"
      />
    </div>
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                The enrollment track determines whether a course run offers a paid certificate and what sort of verification is required.
              </p>
              <p>
                <a
                  href="https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/glossary.html#enrollment-track-g"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Learn more.
                </a>
              </p>
            </div>
          }
          id="run_type.label"
          optional={false}
          required={true}
          text="Course run enrollment track"
        />
      }
      name="run_type"
      options={Array []}
      required={true}
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                Instructor-paced courses include individual assignments that have specific due dates before the course end date.
              </p>
              <p>
                Self-paced courses do not have individual assignments that have specific due dates before the course end date. All assignments are due on the course end date.
              </p>
            </div>
          }
          id="pacing_type.label"
          optional={false}
          text="Course pacing"
        />
      }
      name="pacing_type"
      options={
        Array [
          Object {
            "label": "Instructor-paced",
            "value": "instructor_paced",
          },
          Object {
            "label": "Self-paced",
            "value": "self_paced",
          },
        ]
      }
      type="text"
    />
    <ButtonToolbar
      className=""
      leftJustify={false}
    >
      <Link
        to="/courses/00000000-0000-0000-0000-000000000001"
      >
        <button
          className="btn btn-outline-primary"
          disabled={true}
          type="button"
        >
          Cancel
        </button>
      </Link>
      <ActionButton
        className=""
        disabled={false}
        labels={
          Object {
            "default": "Create",
            "pending": "Creating",
          }
        }
        primary={true}
        state="pending"
      />
    </ButtonToolbar>
  </form>
</div>
`;

exports[`CreateCourseRunForm renders html correctly with Course Type 1`] = `
<div
  className="create-course-run-form"
>
  <h2>
    Create a new course run
  </h2>
  <hr />
  <FieldLabel
    className="mb-2"
    extraText=""
    helpText=""
    id={null}
    optional={false}
    text="Course"
  />
  <div
    className="mb-3"
  >
    <Link
      to="/courses/00000000-0000-0000-0000-000000000001"
    >
      Test Course
    </Link>
  </div>
  <form
    onSubmit={[Function]}
  >
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                Select a run to copy as the starting point for your new studio instance. The newest run is listed first.
              </p>
            </div>
          }
          id="rerun.label"
          optional={false}
          text="Select a run to copy"
        />
      }
      name="rerun"
      options={Array []}
      type="text"
    />
    <div>
      <Field
        component={[Function]}
        dateLabel="Start date"
        helpText={
          <div>
            <p>
              Required Format: yyyy/mm/dd
            </p>
            <p>
              Start on a Tuesday, Wednesday, or Thursday.
            </p>
            <p>
              Avoid major holidays.
            </p>
            <p>
              Dates are editable in Studio after the course is created, rounded to the nearest term. If your courses start on January - May 2020, the URL will end in 1T2020.
            </p>
            <p>
              If you are unsure of the exact date, specify a day that is close to the estimated start date. For example, if your course will start near the end of March, specify March 31.
            </p>
          </div>
        }
        maxLength="10"
        name="start"
        pattern="20[1-9][0-9]/(0[1-9]|1[012])/(0[1-9]|[12][0-9]|3[01])"
        placeholder="yyyy/mm/dd"
        required={true}
        timeLabel="Start time (GMT)"
        type="text"
      />
      <Field
        component={[Function]}
        dateLabel="End date"
        helpText={
          <div>
            <p>
              Required Format: yyyy/mm/dd
            </p>
            <p>
              If you are unsure of the exact date, specify a day that is close to the estimated end date. For example, if your course will end near the end of March, specify March 31.
            </p>
          </div>
        }
        maxLength="10"
        name="end"
        pattern="20[1-9][0-9]/(0[1-9]|1[012])/(0[1-9]|[12][0-9]|3[01])"
        placeholder="yyyy/mm/dd"
        required={true}
        timeLabel="End time (GMT)"
        type="text"
      />
    </div>
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                The enrollment track determines whether a course run offers a paid certificate and what sort of verification is required.
              </p>
              <p>
                <a
                  href="https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/glossary.html#enrollment-track-g"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Learn more.
                </a>
              </p>
            </div>
          }
          id="run_type.label"
          optional={false}
          required={true}
          text="Course run enrollment track"
        />
      }
      name="run_type"
      options={
        Array [
          Object {
            "label": "Select enrollment track",
            "value": "",
          },
          Object {
            "label": "Verified and Audit",
            "value": "4e260c57-24ef-46c1-9a0d-5ec3a30f6b0c",
          },
          Object {
            "label": "Audit Only",
            "value": "cfacfc62-54bd-4e1b-939a-5a94f12fbd8d",
          },
        ]
      }
      required={true}
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                Instructor-paced courses include individual assignments that have specific due dates before the course end date.
              </p>
              <p>
                Self-paced courses do not have individual assignments that have specific due dates before the course end date. All assignments are due on the course end date.
              </p>
            </div>
          }
          id="pacing_type.label"
          optional={false}
          text="Course pacing"
        />
      }
      name="pacing_type"
      options={
        Array [
          Object {
            "label": "Instructor-paced",
            "value": "instructor_paced",
          },
          Object {
            "label": "Self-paced",
            "value": "self_paced",
          },
        ]
      }
      type="text"
    />
    <ButtonToolbar
      className=""
      leftJustify={false}
    >
      <Link
        to="/courses/00000000-0000-0000-0000-000000000001"
      >
        <button
          className="btn btn-outline-primary"
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </Link>
      <ActionButton
        className=""
        disabled={true}
        labels={
          Object {
            "default": "Create",
            "pending": "Creating",
          }
        }
        primary={true}
        state="default"
      />
    </ButtonToolbar>
  </form>
</div>
`;
